butane/tests/json.rs

```diff
@@ -278,3 +278,31 @@ async fn inline_json(conn: ConnectionAsync) {
     let foo3 = OuterFoo::get(&conn, id).await.unwrap();
     assert_eq!(foo2, foo3);
 }
+
+#[butane_test]
+async fn r_hash_field_type_newtype_json(conn: ConnectionAsync) {
+    #[derive(Clone, Debug, Default, FieldType, PartialEq, serde::Serialize, serde::Deserialize)]
+    #[allow(non_camel_case_types)]
+    struct r#type {
+        r#type: String,
+    }
+
+    #[model]
+    #[derive(Debug, Default)]
+    pub struct StructWithReservedWordMember {
+        id: String,
+
+        r#type: r#type,
+    }
+
+    let mut foo = StructWithReservedWordMember::default();
+    foo.id = "1".to_string();
+    let expected = r#type {
+        r#type: "test".to_string(),
+    };
+    foo.r#type = expected.clone();
+    foo.save(&conn).await.unwrap();
```

Comment:
This unwrap on line 304 fails when `struct r#type { ... }` is not `pub`.
```
thread 'r_hash_field_type_newtype_json_async_pg' panicked at butane/tests/json.rs:304:27:
called `Result::unwrap()` on an `Err` value: Postgres(Error { kind: ToSql(1), cause: Some(WrongType { postgres: Text, rust: "serde_json::value::Value" }) })
```

Reply:
ah, that is part of it, but the cause is that butane/ now has a model called `r#type` and a field type called `r#type`